iris_grib/__init__.py

```diff
@@ -1024,7 +1024,7 @@ def save_grib2(cube, target, append=False, **kwargs):
     See also :func:`iris.io.save`.
 
     """
-    messages = as_messages(cube)
+    messages = (message for _, message in save_pairs_from_cube(cube))
```

Comment:
Given that you've changed the behaviour of this function, have you also updated the tests for it?